src/coreclr/src/System.Private.CoreLib/src/System/Reflection/Emit/TypeBuilder.cs

```diff
@@ -415,6 +418,8 @@ internal static unsafe void SetConstantValue(ModuleBuilder module, int tk, Type
         private Type? m_enumUnderlyingType;
         internal bool m_isHiddenGlobalType;
         private bool m_hasBeenCreated;
+
+        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)]
```

Comment:
I assume there's still a warning generated in `CreateTypeNoLock` where the value is assigned - right?

Reply:
The creation happens in a QCall that linker cannot analyze, so it doesn't actually see the assignment and won't warn.